zerver/tests/test_realm_export.py

```diff
@@ -0,0 +1,62 @@
+import mock
+
+from django.utils.timezone import now as timezone_now
+from django.conf import settings
+
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.test_helpers import (
+    use_s3_backend,
+    create_s3_buckets,
+)
+
+from zerver.views.public_export import public_only_realm_export
+from zerver.models import RealmAuditLog
+
+class RealmExportTest(ZulipTestCase):
+    def test_hit_api_endpoint(self) -> None:
+        admin = self.example_user('iago')
+        admin_email = admin.email
+        self.login(admin_email)
+        result = self.client_post('/json/exports/data_export')
+        self.assert_json_success(result)
+        self.logout()
+
+    def test_export_as_not_admin(self) -> None:
+        user = self.example_user('hamlet')
+        not_admin_email = user.email
+        self.login(not_admin_email)
+        with self.assertRaises(JsonableError):
+            public_only_realm_export(self.client_post, user)
+        self.logout()
+
+    @use_s3_backend
+    @mock.patch('zerver.worker.queue_processors.assign_queue')
+    def test_public_only_realm_export(self, mock_worker: mock.MagicMock) -> None:
+        admin = self.example_user('iago')
+        admin_email = admin.email
+        self.login(admin_email)
+        event_time = timezone_now()
+        event_type = RealmAuditLog.REALM_EXPORTED
+        create_s3_buckets(
+            settings.S3_AUTH_UPLOADS_BUCKET,
+            settings.S3_AVATAR_BUCKET)
+
+        mock_worker.event = 'realm_exported'
+        self.assertEqual(mock_worker.event, event_type)
+
+        current_log = RealmAuditLog.objects.filter(event_type=event_type)
+        self.assertEqual(len(current_log), 0)
+
+        exports = []
+        for i in range(0, 5):
+            exports.append(RealmAuditLog(realm=admin.realm,
+                                         realm_id=admin.realm.id,
+                                         event_type=event_type,
+                                         event_time=event_time))
+        RealmAuditLog.objects.bulk_create(exports)
+
+        msg = 'Exceeding weekly organization export limit!'
```

Comment:
again, not worth an extra variable for this -- just pass the string into `assert_json_error` like our other tests do.